spec/requests/providers/statement_of_case_spec.rb

```diff
@@ -66,20 +67,42 @@
 
       it 'updates the record' do
         subject
-        expect(legal_aid_application.statement_of_case.reload.statement).to eq(entered_text)
-        expect(legal_aid_application.statement_of_case.original_files.first).to be_present
+        expect(statement_of_case.reload.statement).to eq(entered_text)
+        expect(statement_of_case.original_files.first).to be_present
       end
 
       it 'redirects to the next page' do
         subject
         expect(response).to redirect_to providers_legal_aid_application_estimated_legal_costs_path(legal_aid_application)
       end
 
+      context '2 files are uploaded' do
+        before do
+          legal_aid_application.build_statement_of_case.save!
+          filepath = "#{Rails.root}/spec/fixtures/files/documents/hello_world.pdf"
+          statement_of_case.original_files.attach(io: File.open(filepath), filename: 'hello_world.pdf', content_type: 'application/pdf')
+          PdfConverterWorker.clear
+        end
+
+        it 'converts both files to pdf' do
+          expect { subject && PdfConverterWorker.drain }.to change { PdfFile.count }.by(2)
+
+          statement_of_case.original_files.each do |original_file|
+            pdf_file = PdfFile.find_by(original_file_id: original_file.id).file
+            expect(pdf_file.filename.base).to eq(original_file.filename.base), 'Should have the same base name'
+          end
+        end
+      end
+
       context 'Continue button pressed' do
         context 'model has no files attached previously' do
           context 'file is empty and text it empty' do
-            before { params[:statement_of_case].delete(:original_files) }
-            let(:entered_text) { '' }
+            let(:params_statement_of_case) do
+              {
+                statement: ''
+              }
```

Comment:
In the context `context 'text is empty but file is present' do` the condition where `statement` is `''` is achieved by `let(:entered_text) { '' }`. I think the same pattern could be used here to keep consistency.